src/utils/httpRequestParser.ts

```diff
@@ -100,7 +114,15 @@ export class HttpRequestParser implements IRequestParser {
         // parse body
         const contentTypeHeader = getHeader(headers, 'content-type') || getHeader(this._restClientSettings.defaultHeaders, 'content-type');
         body = HttpRequestParser.parseRequestBody(bodyLines, requestAbsoluteFilePath, contentTypeHeader);
-        if (this._restClientSettings.formParamEncodingStrategy !== FormParamEncodingStrategy.Never && body && typeof body === 'string' && MimeUtility.isFormUrlEncoded(contentTypeHeader)) {
+        if (isGraphQlRequest) {
+            variables = HttpRequestParser.parseRequestBody(variableLines, requestAbsoluteFilePath, contentTypeHeader);
+
+            let graphQlPayload = {
+                query: body,
+                variables: variables ? JSON.parse(variables.toString()) : {}

```

Comment:
Or the variables should be omitted if variables not provided? I think the variables field is optional

Reply:
Sorry, my comment got deleted. Some GraphQL servers do schema validation checks for `{query,variables}`. I'd rather send it, even if it's empty.

Reply:
Got